lbz/configuration.py

```diff
@@ -0,0 +1,45 @@
+from abc import abstractmethod
+from os import getenv
+from typing import Any, Callable, Generic, Optional, TypeVar
+
+from lbz.aws_boto3 import SSM
+
+T = TypeVar("T")
+
+
+class BaseValue(Generic[T]):
+    def __init__(
+        self,
+        key: str,
+        parser: Optional[Callable[..., T]] = None,
+        default: T = None,
+    ):
+        self.key = key
+        self.parser = parser
+        self.default = default
```

Comment:
I would make all these attributes protected - we don't want to change them from outside anyway:
```suggestion
        self._key = key
        self._parser = parser
        self._default = default
```